### PR TITLE
Alternate spreadsheet

### PR DESCRIPTION
**Pull request by ventusignis**
I put together this PR because I find it hard to easily distinguish full pays from half pays and withholds in the OR history.

This PR adds, as an user setting, an alternate OR history that mimics rr18xx style (|x| for halfpay, [x] for withhold)

image

### CODE REVIEW

File: assets/app/lib/settings.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ module Settings
     SETTINGS = {
       notifications: true,
       simple_logos: false,
+      alternate_spreadsheet: false,
```
Comment (yzemaze) [inline assets/app/lib/settings.rb:24]: No need for an option. The dotted underlining doesn’t really work with linked actions anyway, so please revise this PR, get rid of dotted underlining and implement your changes without user choice.

File: assets/app/view/game/spreadsheet.rb [hunk 2]
```diff
@@ -121,38 +122,84 @@ def render_history(corporation)
...
+          props = {
+            style: {
+              opacity: case hist[x].dividend.kind
+                       when 'withhold'
+                         '0.5'
+                       when 'half'
+                         '0.75'
+                       else
+                         '1.0'
+                       end,
```
Comment (yzemaze) [inline assets/app/view/game/spreadsheet.rb:154]: One case-statement with two return values would be nice.

**Review by yzemaze — CHANGES_REQUESTED** [review]

File: assets/app/lib/settings.rb (hunk 1 above)
Comment (ventusignis) [reply assets/app/lib/settings.rb]: I removed the choice to use the old style

File: assets/app/view/game/spreadsheet.rb (hunk 2 above)
Comment (ventusignis) [reply assets/app/view/game/spreadsheet.rb]: I was able to pull that off

File: assets/app/view/game/spreadsheet.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@ class Spreadsheet < Snabberb::Component
       include Actionable
 
       needs :game
+      needs :user, default: nil, store: true
```
Comment (yzemaze) [inline assets/app/view/game/spreadsheet.rb:19]: still necessary? (btw: why store?)

**Comment by yzemaze** [comment]
I just checked out your branch and after seeing it live I’m not really convinced by using | | for half-pay. The visual difference to [ ] is just to minuscule. What do you think about using : : instead?
image

What’s your take on the underlined links in general? Any way we could get rid of the visual clutter but stil convey these are links to actions? It would look a lot cleaner …
image

**Comment by ventusignis** [comment]
@yzemaze What if we removed the underlines and swapped in blue text?
The colons look good. I used |x| because that's what rr18xx used for half pays

**Comment by yzemaze** [comment]
> What if we removed the underlines and swapped in blue text?
I hate blue text and it collides with opacity and user defined bg color.

> The colons look good. I used |x| because that's what rr18xx used for half pays
Ok, then apply : and please check the other comment re: `needs: user`.

**Comment by daniel-sousa-me** [comment]
> I hate blue text and it collides with opacity and user defined bg color.

This. Plus using color to convey information is not color-blind-friendly (8% of male population)

I'm all for removing underlines. Underlining is generally considered bad typography.
I'm conflicted about which chars to use. I like the idea of being consistent with rr18xx, but I agree that the square brackets and the pipes are not very easy to distinguish in the first image. But maybe it becomes better without the underlines? (plus, there's a difference in opacity to help). I personally don't like the colons there aesthetically.

Since our unicode support is pretty consistent, we can also make use of that. Instead of using the ascii pipe, we could use some other unicode char that looks similar, but farther from brackets (longer, shorter, broken, double, thicker, thinner...)

**Comment by yzemaze** [comment]
> I'm all for removing underlines.

Well, do we need to highlight the fact that these are links or could we just get rid of the underlines and rely on (generally savvy) spreadsheet users knowing this by now and others just discovering it? Maybe adding a title/tooltip to the header would help. btw: Linking round numbers to actions might be useful, too.

> Instead of using the ascii pipe, we could use some other unicode char that looks similar, but farther from brackets (longer, shorter, broken, double, thicker, thinner...)

Broken bar (U+000A6) looks indeed nicer than colon:
image

**Comment by ventusignis** [comment]
I've uploaded changes
image

File: assets/app/view/game/spreadsheet.rb (hunk 3 above)
Comment (ventusignis) [reply assets/app/view/game/spreadsheet.rb]: No, removed

**Review by yzemaze — APPROVED** [review]

**Pull request merged**